plugins/node/opentelemetry-instrumentation-bunyan/src/instrumentation.ts

```diff
@@ -14,54 +14,88 @@
  * limitations under the License.
  */
 
-import {
-  context,
-  diag,
-  trace,
-  isSpanContextValid,
-  Span,
-} from '@opentelemetry/api';
+import { inherits } from 'util';
+import { context, trace, isSpanContextValid, Span } from '@opentelemetry/api';
 import {
   InstrumentationBase,
   InstrumentationNodeModuleDefinition,
-  isWrapped,
   safeExecuteInTheMiddle,
 } from '@opentelemetry/instrumentation';
 import { BunyanInstrumentationConfig } from './types';
 import { VERSION } from './version';
+import { OpenTelemetryBunyanStream } from './OpenTelemetryBunyanStream';
 import type * as BunyanLogger from 'bunyan';
 
+const DEFAULT_CONFIG: BunyanInstrumentationConfig = {
+  enableLogsBridge: true,
+  enableInjection: true,
+};
+
 export class BunyanInstrumentation extends InstrumentationBase<
   typeof BunyanLogger
 > {
   constructor(config: BunyanInstrumentationConfig = {}) {
-    super('@opentelemetry/instrumentation-bunyan', VERSION, config);
+    super(
+      '@opentelemetry/instrumentation-bunyan',
+      VERSION,
+      Object.assign({}, DEFAULT_CONFIG, config)
+    );
   }
 
   protected init() {
     return [
       new InstrumentationNodeModuleDefinition<typeof BunyanLogger>(
         'bunyan',
         ['<2.0'],
-        logger => {
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          const proto = logger.prototype as any;
-          if (isWrapped(proto['_emit'])) {
-            this._unwrap(proto, '_emit');
-          }
+        (module: any, moduleVersion) => {
+          this._diag.debug(`Applying patch for bunyan@${moduleVersion}`);
+          const instrumentation = this;
+          const Logger =
+            module[Symbol.toStringTag] === 'Module'
+              ? module.default // ESM
+              : module; // CommonJS
 
           this._wrap(
-            proto,
+            Logger.prototype,
             '_emit',
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             this._getPatchedEmit() as any
           );
-          return logger;
-        },
-        logger => {
-          if (logger === undefined) return;
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          this._unwrap(logger.prototype as any, '_emit');
+
+          function LoggerTraced(this: any, ...args: unknown[]) {
+            let inst;
+            let retval = undefined;
+            if (this instanceof LoggerTraced) {
+              // called with `new Logger()`
+              inst = this;
+              Logger.call(this, ...args);
+            } else {
+              // called without `new`
+              inst = Logger.call(null, ...args);
```

Comment:
commit 7d1532fb

Reply:
Heh, now I recall why I used `.call(null, ...args)` -- because I get a lint error with `.apply`:
https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/6805430162/job/18504931484?pr=1713

```
 /__w/opentelemetry-js-contrib/opentelemetry-js-contrib/plugins/node/opentelemetry-instrumentation-bunyan/src/instrumentation.ts
   50:18  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
   65:39  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
   74:22  error    Use the spread operator instead of '.apply()'  prefer-spread
  154:30  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any

✖ 5 problems (1 error, 4 warnings)
```

Reply:
commit 44d20bde